packages/cells/src/widget.ts

```diff
@@ -1098,6 +1101,70 @@ export class CodeCell extends Cell<ICodeCellModel> {
     model.stateChanged.connect(this.onStateChanged, this);
   }
 
+  /**
+   * Observe the movement of the caret in the output area.
+   *
+   * Emits scroll request if the caret moved.
+   */
+  private _observeCaretMovementInOuput(output: OutputArea) {
+    let lastOnCaretMovedHandler: () => void;
+    let lastTarget: HTMLElement | null;
+
+    output.node.addEventListener('keydown', (e: KeyboardEvent) => {
+      const inWindowedContainer = this._inViewport !== null;
+      const defaultPrevented = inWindowedContainer && !this._inViewport;
+
+      // Because we do not want to scroll on any key, but only on keys which
+      // move the caret (this on keys which cause input and on keys like left,
+      // right, top, bottom arrow, home, end, page down/up - but only if the
+      // cursor is not at the respective end of the input) we need to listen
+      // to the `selectionchange` event on target inputs/textareas, etc.
+      const target = e.target;
+
+      if (!target || !(target instanceof HTMLElement)) {
+        return;
+      }
+
+      // Make sure the previous listener gets disconnected
+      if (lastTarget) {
+        lastTarget.removeEventListener(
+          'selectionchange',
+          lastOnCaretMovedHandler
+        );
+        document.removeEventListener(
+          'selectionchange',
+          lastOnCaretMovedHandler
+        );
+      }
+
+      const onCaretMoved = () => {
+        this._scrollRequested.emit({
+          scrollWithinCell: ({ scroller }) => {
+            ElementExt.scrollIntoViewIfNeeded(scroller, target);
+          },
+          defaultPrevented
+        });
```

Comment:
Shouldn't the default scrollRequest be emitted when this gets disconnected?

Reply:
I do not understand, can you clarify?

Reply:
Nevermind I read the code too quickly and I had the impression that the callback `scrollWithinCell` was stored in the notebook. But it is not.

Sorry for the noise.